addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -684,6 +691,10 @@ public void scan(HttpMessage msg, String param, String value) {
                                                             .getResponseHeader()
                                                             .getHeader(HttpHeader.CONTENT_TYPE),
                                                     "json")) {
+                                                /* avoid sending a null byte to ZAP UI: */        
+                                                if (scriptAlert == GENERIC_NULL_BYTE_SCRIPT_ALERT) {
+                                                    scriptAlert = GENERIC_SCRIPT_ALERT;
+                                                }
```

Comment:
Was this actually a problem or are you just choosing to be extra careful?

Reply:
extra careful.